Rebus.LegacyExtensions/LegacyCompatibility/CustomJsonSerializer.cs

```diff
@@ -0,0 +1,165 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+
+using Rebus.Extensions;
+using Rebus.Messages;
+using Rebus.Serialization;
+
+#pragma warning disable 1998
+
+namespace Rebus.LegacyCompatibility
```

Comment:
Please, decide between Rebus.LegacyCompatibility vs Rebus.LegacyExtensions.. From what I see there is already a LegacyCompatibility project at upstream, so we _should_ avoid using the same name/terms. So please, rename everything as Rebus.LegacyExtensions, or equivalent as to avoid confusions.